File: go/0538M-把二叉搜索树转换为累加树.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-12 11:38:39
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func convertBST(root *TreeNode) *TreeNode {
	sum := 0
	var dfs func(node *TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}
		dfs(node.Right)
		node.Val += sum
		sum = node.Val
		dfs(node.Left)
	}
	dfs(root)
	return root
}

File: cpp/0538M-把二叉搜索树转换为累加树.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-23 19:15:26
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int sum = 0;
    TreeNode* convertBST(TreeNode* root) {
        if (root == nullptr) return nullptr;
        root->right = convertBST(root->right);
        root->val = root->val + sum;
        sum = root->val;
        root->left = convertBST(root->left);
        return root;
    }
};